File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/monitoring/port/PerformanceMetricsRepository.java
package com.study.webflux.rag.domain.monitoring.port;

import java.time.Instant;

import com.study.webflux.rag.domain.monitoring.model.PerformanceMetrics;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PerformanceMetricsRepository {
	Mono<PerformanceMetrics> save(PerformanceMetrics metrics);

	Mono<PerformanceMetrics> findById(String pipelineId);

	Flux<PerformanceMetrics> findByTimeRange(Instant startTime, Instant endTime);

	Flux<PerformanceMetrics> findByStatus(String status, int limit);

	Flux<PerformanceMetrics> findSlowPipelines(long thresholdMillis, int limit);

	Flux<PerformanceMetrics> findRecent(int limit);
}

File: domain/src/main/kotlin/com/miyou/app/domain/monitoring/port/PerformanceMetricsRepository.kt
package com.miyou.app.domain.monitoring.port

import com.miyou.app.domain.monitoring.model.PerformanceMetrics
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.time.Instant

interface PerformanceMetricsRepository {
    fun save(metrics: PerformanceMetrics): Mono<PerformanceMetrics>

    fun findById(pipelineId: String): Mono<PerformanceMetrics>

    fun findByTimeRange(
        startTime: Instant,
        endTime: Instant,
    ): Flux<PerformanceMetrics>

    fun findByStatus(
        status: String,
        limit: Int,
    ): Flux<PerformanceMetrics>

    fun findSlowPipelines(
        thresholdMillis: Long,
        limit: Int,
    ): Flux<PerformanceMetrics>

    fun findRecent(limit: Int): Flux<PerformanceMetrics>
}
